pkg/component/kubernetes/controllermanager/controllermanager.go

```diff
@@ -476,102 +453,34 @@ func (k *kubeControllerManager) Deploy(ctx context.Context) error {
+		return err
+	}
 
-		var (
-			updateModeAuto = hvpav1alpha1.UpdateModeAuto
-			vpaLabels      = map[string]string{v1beta1constants.LabelRole: "kube-controller-manager-vpa"}
-		)
+	vpaUpdateMode := vpaautoscalingv1.UpdateModeAuto
 
-		scaleDownUpdateMode := k.values.HVPAConfig.ScaleDownUpdateMode
-		if scaleDownUpdateMode == nil {
-			scaleDownUpdateMode = ptr.To(hvpav1alpha1.UpdateModeAuto)
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, k.seedClient.Client(), vpa, func() error {
+		vpa.Spec.TargetRef = &autoscalingv1.CrossVersionObjectReference{
+			APIVersion: appsv1.SchemeGroupVersion.String(),
+			Kind:       "Deployment",
+			Name:       k.values.NamePrefix + v1beta1constants.DeploymentNameKubeControllerManager,
 		}
-
-		if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, k.seedClient.Client(), hvpa, func() error {
-			hvpa.Labels = utils.MergeStringMaps(
-				hvpa.Labels,
-				getLabels(),
-				map[string]string{
-					resourcesv1alpha1.HighAvailabilityConfigType: resourcesv1alpha1.HighAvailabilityConfigTypeController,
-				},
-			)
-			hvpa.Spec.Replicas = ptr.To[int32](1)
-			hvpa.Spec.Hpa = hvpav1alpha1.HpaSpec{
-				Deploy:   false,
-				Selector: &metav1.LabelSelector{MatchLabels: getLabels()},
-				Template: hvpav1alpha1.HpaTemplate{
-					ObjectMeta: metav1.ObjectMeta{
-						Labels: getLabels(),
-					},
-					Spec: hvpav1alpha1.HpaTemplateSpec{
-						MinReplicas: ptr.To[int32](1),
-						MaxReplicas: int32(1),
-					},
-				},
-			}
-			hvpa.Spec.Vpa = hvpav1alpha1.VpaSpec{
-				Selector: &metav1.LabelSelector{MatchLabels: vpaLabels},
-				Deploy:   true,
-				ScaleUp: hvpav1alpha1.ScaleType{
-					UpdatePolicy: hvpav1alpha1.UpdatePolicy{
-						UpdateMode: &updateModeAuto,
-					},
-				},
-				ScaleDown: hvpav1alpha1.ScaleType{
-					UpdatePolicy: hvpav1alpha1.UpdatePolicy{
-						UpdateMode: scaleDownUpdateMode,
-					},
-				},
-				Template: hvpav1alpha1.VpaTemplate{
-					ObjectMeta: metav1.ObjectMeta{
-						Labels: vpaLabels,
-					},
-					Spec: hvpav1alpha1.VpaTemplateSpec{
-						ResourcePolicy: hvpaResourcePolicy,
-					},
-				},
-			}
-			hvpa.Spec.WeightBasedScalingIntervals = []hvpav1alpha1.WeightBasedScalingInterval{
-				{
-					VpaWeight:         hvpav1alpha1.VpaOnly,
-					StartReplicaCount: 1,
-					LastReplicaCount:  1,
-				},
-			}
-			hvpa.Spec.TargetRef = &autoscalingv2beta1.CrossVersionObjectReference{
-				APIVersion: appsv1.SchemeGroupVersion.String(),
-				Kind:       "Deployment",
-				Name:       k.values.NamePrefix + v1beta1constants.DeploymentNameKubeControllerManager,
-			}
-			return nil
-		}); err != nil {
-			return err
-		}
-	} else {
-		if err := kubernetesutils.DeleteObject(ctx, k.seedClient.Client(), hvpa); err != nil {
-			return err
+		vpa.Spec.UpdatePolicy = &vpaautoscalingv1.PodUpdatePolicy{
+			UpdateMode: &vpaUpdateMode,
 		}
+		vpa.Spec.ResourcePolicy = vpaResourcePolicy
```

Comment:
Variable is not needed. I was only keeping it in case we want it for consistency with some other code.

Removed variable and inlined object literal.